### DISCUSSION TITLE
How to get identifyTissueOutline to change to fit image

### DISCUSSION THREAD

**Discussion by alosdiallo** [theMILOlab/SPATA2#164] [Q&A]
I am trying to get identifyTissueOutline to work with my image. I have tried to play with the values a lot but have not had any success. The mask works perfectly as does plotImage with the outline.  I'm attaching images of what the results look like. The  code is pretty basic:
`object <- identifyPixelContent(object)

plotImageMask(object)

plotPixelContent(object)

plotImage(object) + 
  ggpLayerFrameByCoords(object)
plotImage(object, outline = TRUE, line_size = 1)

object <- identifyTissueOutline(object)
plotSurface(object, color_by = "tissue_section" ,pt_alpha = 0.4)`

Any advice would be wonderful. Thank you!
![plotSurface](https://github.com/user-attachments/assets/bc4773b6-2c53-42a6-a0c3-ca1965d1c118)
![maskoutlineImage](https://github.com/user-attachments/assets/22209a07-c891-4383-a3ee-a706aef06939)
![plotImage_Cropped](https://github.com/user-attachments/assets/86b2b056-5467-4282-b547-97437e684bf9)
![plotPixelContent](https://github.com/user-attachments/assets/fd9078d6-ddd2-4c83-a201-345cc7e79c97)
![plotImageMask](https://github.com/user-attachments/assets/114e1561-71c5-4cd6-84a6-0cca6c914552)

**Comment by kueckelj** [comment]
Yes, this is a problem of concaveman() not being precise enough. Sometimes it takes "shortcuts". I think that using an image of higher resolution helps here, but it seems that you are already using image "hires", right?

Unrelatd: I notice that your grid of spots includes all spots of the platform while many of them do not seem to have any tissue underlying them. Do they all have counts? Or is that another bug I have to fix?

**Reply by kueckelj** [reply]
Or do you want to subset spots based on the outline created with the image? If so, you can try `object <- identifySpatialOutliers(object, method = 'image')` and the visualize the results with `plotSurface(object, color_by = 'sp_outlier')`. If that fits your expectations you can subset the object afterwards with `object_sub <- removeSpatialOutliers(object)`. This essentially defines and removes spots that are not located within the outline defined by `identifyTissueOutline(..., method = 'image')`.

**Comment by alosdiallo** [comment]
Hi Jan,
Yes I am using hires. That helped a lot! I don't think this is a bug for you to fix.  I didn't make the selection of the capture area someone in pathology shared resource did. I do think though It looks as good as it's going to look.  object_sub gives me what I am looking for.  One remaining question is, I have two tissues on the same visium capture area.  Is it possible to separate them given how close they are, I think I can do this manually as I did with createSpatialSegmentation? This is less of a concern but would be nice.  Thank you! 
<img width="371" alt="CytAssist Image Alignment" src="https://github.com/user-attachments/assets/2d0ed441-7d1f-42ed-981c-c302ff120ab3">
<img width="462" alt="Tissue Detection and Fiducial Alignment" src="https://github.com/user-attachments/assets/28990c4a-4f0b-496e-b16d-4f762f927cf5">
<img width="416" alt="Visium_Image" src="https://github.com/user-attachments/assets/c8273e5f-7e80-4f15-824b-a21085ff2bdc">

![Refined_outline](https://github.com/user-attachments/assets/fc4e7b60-59fe-4a72-a46b-83977d31480b)

**Comment by kueckelj** [comment]
Well, you can play around with the parameters of `identifyPixelContent(..., method = "image")`, group the spots based on that outline once it suits your expectations and go with `splitSpataObject()`. To save time, I think it would indeed be easiest to quickly group the spots with `createSpatialSegmentation()` and then use `splitSpataObject()`.

**Comment by alosdiallo** [comment]
Thanks Jan!  I'm away for a little bit so I will give that a try once I'm
back.  I really appreciate you helping me out with this! Thank you!
